cmd/init-virtual-kubelet/main.go

```diff
@@ -88,8 +88,7 @@ func main() {
 	cancel()
 
 	if err != nil {
-		klog.Error("Unable to get certificate: %w", err)
-		return
+		klog.Fatalf("Unable to get certificate: %w", err)
```

Comment:
I'm afraid that this can break init on EKS, since the certificate is not provided in the CSR.

Reply:
I personally changed that to prevent silent failures (as it was happening before the fix), but I can revert if it creates problems on EKS

Reply:
I updated the PR to add the possibility to disable the init-container. Let me know if this is what you meant.

Reply:
lgtm